packages/tables/src/Concerns/CanSelectRecords.php

```diff
@@ -45,7 +45,16 @@ public function getAllTableRecordsCount(): int
             return $this->records->total();
         }
 
-        return $this->getFilteredTableQuery()->count();
+        $query = $this->getFilteredTableQuery();
+
+        if (is_callable($this->isRecordSelectable())) {
+            return $query
+                ->get()
+                ->filter(fn ($item): bool => $this->isRecordSelectable()($item))
```

Comment:
Does this work if we make it a little more explicit? Might have to add the `Model` import.

```suggestion
                ->filter(fn (Model $record): bool => $this->isRecordSelectable()($record))
```